nilearn/image/resampling.py

```diff
@@ -474,6 +467,18 @@ def resample_img(img, target_affine=None, target_shape=None,
     else:
         resampled_data_dtype = data.dtype
 
+    # ndimage.affine_transform which is in helper function _resample_one_img
+    # does not handle well data not in native endianness.
+    # Hence, we swap new byte order to change dtype to native endianness.
+    # see https://github.com/nilearn/nilearn/issues/275 &
+    # https://github.com/nilearn/nilearn/issues/1445
+    # This bug is supposed to be fixed in Scipy 0.15 but it is not. Hence we
+    # are not checking any Scipy versions here.
+
+    # Byte swap here, if it is big-endian
+    if not resampled_data_dtype.isnative:
+        resampled_data_dtype = resampled_data_dtype.newbyteorder()
```

Comment:
It should be possible to detect whether or not endianness is swapped. I'll take a look into it also if I have time this evening.